src/main/scala/io/iohk/ethereum/db/dataSource/DataSourceBatchUpdate.scala

```diff
@@ -0,0 +1,17 @@
+package io.iohk.ethereum.db.dataSource
+
+case class DataSourceBatchUpdate(dataSource: DataSource, updates: Seq[DataSourceUpdate] = Nil) {
+
+  def and(that: DataSourceBatchUpdate): DataSourceBatchUpdate = {
+    require(
+      this.dataSource eq that.dataSource,
+      "Transactional storage updates must be performed on the same data source"
+    )
+    DataSourceBatchUpdate(dataSource, this.updates ++ that.updates)
```

Comment:
I am wondering if queue with appending or list with prepending would not be better here ? As otherwise when batching a lot of updates from list (like in FastSync) we have quite an overhead

Reply:
Used Array based on https://www.lihaoyi.com/post/BenchmarkingScalaCollections.html#concatenation-performance